src/main/java/org/broadinstitute/hellbender/Main.java

```diff
@@ -91,6 +93,36 @@ protected static void printDecoratedExceptionMessage(final PrintStream ps, final
         return packageList;
     }
 
+
+    /**
+     * Reads from the given command-line arguments, pulls out configuration options,
+     * and initializes the configuration for this instance of Main.
+     * Returns the list of arguments without the config file option and config file path.
+     *
+     * Suggested use for this is to handle downstream project configuration options and overrides.
+     * For example this would allow:
+     *
+     *      Custom command-line arguments for use in tools
+     *      Custom config file loading and initialization
+     *
+     * @param args The Array of command-line arguments.
+     * @return A new Array of command-line arguments with the config file option and config file path removed.
+     */
+    protected String[] parseArgsForConfigSetupAndGetNewArgs(final String[] args) {
+        // First we package our args together a little better
+        // so that we can pull out the argument for the config file (and the config filename itself)
+        // prior to the args being processed by extractCommandLineProgram.
+        final ArrayList<String> argArrayList = new ArrayList<>( Arrays.asList(args) );
+
+        // Now we setup our configurations:
+        // This method will modify the given argArrayList to remove the gatkConfigFileOption and value
+        // if they are specified.  This is so that the later validation of commandline arguments can still
+        // happen properly and so that we can have the configuration initialized as early as possible.
+        ConfigUtils.initializeConfigurationsFromCommandLineArgs(argArrayList, StandardArgumentDefinitions.GATK_CONFIG_FILE_OPTION);
```

Comment:
Can we make this pure and just return the updated list instead of modifying the input (same for the underlying methods) ?